File: frontend/src/types/substance.ts
export interface Substance {
  formula: string;
  name: string;
  phase: "s" | "l" | "g" | "aq";
  color: string;
  amount_ml: number | null;
  concentration: number | null;
  molar_mass: number | null;
  hazard_class: string | null;
}

File: backend/app/models/substance.py
from pydantic import BaseModel


class Substance(BaseModel):
    formula: str
    name: str
    phase: str  # "s", "l", "g", "aq"
    color: str
    amount_ml: float | None = None
    concentration: float | None = None
    molar_mass: float | None = None
    hazard_class: str | None = None


class SubstanceInput(BaseModel):
    formula: str
    amount_g: float | None = None
    amount_ml: float | None = None
    phase: str
